readthedocs/rtd_tests/tests/test_project_forms.py

```diff
@@ -333,6 +335,55 @@ def test_trigger_build_on_save(self, trigger_build):
             version=default_branch,
         )
 
+    def test_set_remote_repository(self):
+        data = {
+            "name": "Project",
+            "repo": "https://github.com/readthedocs/readthedocs.org/",
+            "repo_type": self.project.repo_type,
+            "default_version": LATEST,
+            "language": self.project.language,
+            "versioning_scheme": self.project.versioning_scheme,
+        }
+
+        remote_repository = get(
+            RemoteRepository,
+            full_name="rtfd/template",
+            clone_url="https://github.com/rtfd/template",
+            html_url="https://github.com/rtfd/template",
+            ssh_url="git@github.com:rtfd/template.git",
+            private=False,
+        )
+
+        # No remote repository attached.
+        form = UpdateProjectForm(data, instance=self.project, user=self.user)
+        self.assertTrue(form.is_valid())
+
+        # Remote repository attached, but it doesn't belong to the user.
+        data["remote_repository"] = remote_repository.pk
+        form = UpdateProjectForm(data, instance=self.project, user=self.user)
+        self.assertFalse(form.is_valid())
+        self.assertIn("remote_repository", form.errors)
```

Comment:
Ah so, you added the addition in the widget query for showing the remote repo even if the user doesn't have access to it, but it seems this user might be blocked from updating the form at all in this case too?

Like, if a user without access to the project's remote repository wants to change another field in the form?

Reply:
> Like, if a user without access to the project's remote repository wants to change another field in the form?

No, the current remote repository is included in the queryset itself, so it's an allowed value when saving the form.